fdbserver/VersionIndexerInterface.actor.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * VersionIndexerInterface.actor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if defined(NO_INTELLISENSE) && !defined(FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_G_H)
+#define FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_G_H
+#include "fdbserver/VersionIndexerInterface.actor.g.h"
+#elif !defined(FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_H)
+#include "fdbrpc/fdbrpc.h"
+
+struct VersionIndexerInterface {
+	constexpr static FileIdentifier file_identifier = 6865162;
+
+	UID id = deterministicRandom()->randomUniqueID();
+	RequestStream<ReplyPromise<Void>> waitFailure;
+	RequestStream<struct CommitRequest> commit;
+	RequestStream<struct VersionIndexerPeekRequest> peek;
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, waitFailure, commit, peek);
+	}
+};
+
+struct CommitRequest {
+	constexpr static FileIdentifier file_identifier = 7036778;
+
+	Version version, previousVersion, committedVersion;
+	std::vector<Tag> tags;
+	ReplyPromise<Void> reply;
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, version, previousVersion, committedVersion, tags, reply);
+	}
+};
+
+struct VersionIndexerPeekReply {
+	constexpr static FileIdentifier file_identifier = 10553306;
+
+	Version committedVersion, previousVersion;
+	std::vector<std::pair<Version, bool>> versions;
```

Comment:
Is there going to be an entry in this vector for each committed version in the range (lastKnownVersion, committedVersion]?

Reply:
Thinking more about it, not sure if the version indexer needs to return information about versions to this granularity. Isn't it enough if it returns the following:
- the highest commit version till which the version indexer has entries in its queue (VersionIndexer::version.get())
-  bool: that tells the storage whether it should do a peek from its tlog or not.  If false, the storage doesn't need to peek, and can advance its commit version to "VersionIndexer::version.get()".

Reply:
You can ignore my above comment. I see that you use a stream interface to propagate this information In your later commit.